# Running Assembly Using Megahit

In [6]:
![ -e 'Out Megahit' ] && rm -r 'Out Megahit'
!mkdir 'Out Megahit'

!cd 'Out Megahit' && /media/anuradhawick/data/Tools/megahit/megahit \
-r '/media/anuradhawick/data/Experiments/Assembly Graph/Sampled Reads/all-in-one.fastq.gz'

15.532Gb memory in total.
Using: 13.978Gb.
MEGAHIT v1.1.3
--- [Mon Nov  5 16:16:16 2018] Start assembly. Number of CPU threads 8 ---
--- [Mon Nov  5 16:16:16 2018] Available memory: 16676945920, used: 15009251328
--- [Mon Nov  5 16:16:16 2018] Converting reads to binaries ---
b'    [read_lib_functions-inl.h  : 209]     Lib 0 (/media/anuradhawick/data/Experiments/Assembly Graph/Sampled Reads/all-in-one.fastq.gz): se, 76831 reads, 29879 max length'
b'    [utils.h                   : 126]     Real: 15.9056\tuser: 4.3336\tsys: 0.5093\tmaxrss: 134956'
--- [Mon Nov  5 16:16:32 2018] k list: 21,29,39,59,79,99,119,141 ---
--- [Mon Nov  5 16:16:32 2018] Extracting solid (k+1)-mers for k = 21 ---
--- [Mon Nov  5 16:17:33 2018] Building graph for k = 21 ---
--- [Mon Nov  5 16:17:36 2018] Assembling contigs from SdBG for k = 21 ---
--- [Mon Nov  5 16:17:44 2018] Local assembling k = 21 ---
--- [Mon Nov  5 16:18:10 2018] Extracting iterative edges from k = 21 to 29 ---
--- [Mon Nov  5 16:18:28 2018

# Creating Contig Graphs

In [46]:
from pathlib import Path
import csv

pathlist = [str(path) for path in Path('./Out Megahit/megahit_out/intermediate_contigs/').glob('**/*.contigs.fa')]

for path in pathlist:
    if 'final' in path: continue
    kmer_size = path.split('/')[::-1][0].split('.')[0][1::]
    get_ipython().system("cd 'Out Megahit' && /media/anuradhawick/data/Tools/megahit/megahit_toolkit contig2fastg " + kmer_size + " " + "/".join(path.split('/')[1::]) +" > k" + kmer_size +".contigs.fastg")
    get_ipython().system("cd 'Out Megahit' && /media/anuradhawick/data/Tools/Bandage/Bandage_Ubuntu_dynamic_v0_8_1/Bandage image k" + kmer_size + ".contigs.fastg ../image_graph_megahit_k" + kmer_size +".png")

## Graph k=21
![title](image_graph_megahit_k21.png)
## Graph k=29
![title](image_graph_megahit_k29.png)
## Graph k=39
![title](image_graph_megahit_k39.png)
## Graph k=59
![title](image_graph_megahit_k59.png)
## Graph k=79
![title](image_graph_megahit_k79.png)
## Graph k=99
![title](image_graph_megahit_k99.png)
## Graph k=119
![title](image_graph_megahit_k119.png)
## Graph k=141
![title](image_graph_megahit_k141.png)